### PR TITLE
Deploy to cdn

### PR DESCRIPTION
**Pull request by RamanaChowdam**

### CODE REVIEW

**Comment by krishnaraochintu** [comment]
Looks Good

**Comment by krishnaraochintu** [comment]
Venu, please review now

**Review by jvenugopal — APPROVED** [review]

File: learn/app-development/deployment/app-integration-with-aws-cdn.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+```shell
+mvn clean install -P<profile-name> -Dcdn-url=<CDN_URL>
+```
+
+```shell
+mvn clean install -Pdeployment -Dcdn-url=https://mydomain.cloudfront.net/my_app>/1234/
+```
+
+- The above command will generate a new folder called **target** in the project sources & generates `project.war` file and `ui-artifacts.zip` file in it. The `ui-artifacts.zip` file have static files of the application, You need to unzip the file `ui-artifacts.zip` and upload it to Amazon S3 bucket that we created in the previous steps. Use the following commands for unzip and upload to S3.
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/app-development/deployment/app-integration-with-aws-cdn.md:86]: Replace the line with this:

The above command generates two deployable artifacts: `ui-artifacts.zip`, `project.war`. Both these files can be found in `target` folder.

Also replace this sentence:

Use the following commands for unzip and upload to S3.
with
Use the following commands to unzip and upload to S3.

Note that the correct proposition to use here is "to" not "for"

File: learn/app-development/deployment/app-integration-with-aws-cdn.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+- For unzip the file to a specific folder
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/app-development/deployment/app-integration-with-aws-cdn.md:88]: Wrong. Replace with:

To unzip the file and store contents in a specific folder

File: learn/app-development/deployment/app-integration-with-aws-cdn.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  
+```shell
+unzip ui-artifacts.zip -d <my-static-content-folder>
+```
+
+- For upload files to AWS S3 bucket
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/app-development/deployment/app-integration-with-aws-cdn.md:94]: Upload files to AWS S3 bucket

File: learn/app-development/deployment/app-integration-with-azure-cdn.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+[![CDN endpoint](/learn/assets/wme-setup/cdn-endpoint-creation.png)](/learn/assets/wme-setup/cdn-endpoint-creation.png)
+
+- Notedown the endpoint hostname to provide to maven command for integration
+  - example endpoint hostname
+
+    ```bash
+        https://wavemaker-app-cdn-integration.azureedge.net
+    ```
+
+## WaveMaker App Build Process for CDN support
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/app-development/deployment/app-integration-with-azure-cdn.md:63]: Please don't repeat this content here. Just link to it.

File: learn/app-development/deployment/building-with-maven.md [hunk 5]
```diff
@@ -3,64 +3,104 @@ title: "Building Project with Maven"
...
+mvn clean install -Pdeployment
 ```
 
-![mvn install](/learn/assets/enter-mvn-install-cmd-and-profilename.png)
+- This command will generate a `project war`  file in the target directory. The `project war`  file has both frontend artifacts (HTML, CSS, JS, images, etc), and backend artifacts (Java Classes).
+- This war file can be deployed into any webserver like Tomcat. For more information, see [App Deployment to Tomcat](/learn/how-tos/wavemaker-application-deployment-tomcat).
+
+## Build War file and Static Content to Deploy them Separately
+
+- WaveMaker app is consists of frontend artifacts (HTML, CSS, JS, images, etc), and backend artifacts (Java Classes). It is recommended to host frontend artifacts in Static Content Servlet like nginx, apache, etc, or Content Delivery Network (CDN), and backend artifacts can be hosted on any web server like Tomcat.
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/app-development/deployment/building-with-maven.md:71]: wrong.

WaveMaker app consists of

not 
is consists of

Comment (prasanth-reddy-wavemaker-com) [inline learn/app-development/deployment/building-with-maven.md:71]: Also 

static content server like nginx

not 

static content **servlet** like nginx

File: learn/on-premise/aws/install-prerequisites.md [hunk 6]
```diff
@@ -30,27 +42,36 @@ sudo apt-get install wget  -y
...
 
+- Install Docker Engine 18.06.2
+
 ```bash
-    apt-get install docker-ce=18.06.2~ce~3-0~ubuntu
+    sudo apt-get install docker-ce=18.06.2~ce~3-0~ubuntu -y
 ```
 
-### Extra configurations on Ubuntu StudioWorkspace Instance / AppDeployment Instance
+- For upgrade and install the latest version of Docker
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/on-premise/aws/install-prerequisites.md:61]: To upgrade 

not

**For** upgrade

File: learn/on-premise/aws/install-prerequisites.md [hunk 7]
```diff
@@ -30,27 +42,36 @@ sudo apt-get install wget  -y
...
+  - Run the following command for list available versions
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/on-premise/aws/install-prerequisites.md:62]: Run the following command to list available versions

File: learn/on-premise/aws/install-prerequisites.md [hunk 8]
```diff
@@ -69,10 +90,37 @@ sudo apt-get install python3 -y
 
 ## RHEL
 
-Install below softwares on Platform Instance and StudioWorkspace Instance / AppDeployment Instance
+### The ssh user has privileges(root/sudo) or user don't have privileges for install/upgrade utility softwares
+
+#### Platform Instance
+
+- If given ssh user has privileges(root/sudo) or user don't have privileges to install/upgrade. WME Installer will automatically install the Docker software. Internet is not required for Installation in this case.
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/on-premise/aws/install-prerequisites.md:97]: wrong sentence. please rewrite this

SSH user performing the install or upgrade needs root/sudo privileges.

File: learn/on-premise/aws/install-prerequisites.md [hunk 9]
```diff
@@ -101,33 +149,65 @@ Use same version numbers as mentioned.
...
-### Extra configurations on RHEL StudioWorkspace Instance / AppDeployment Instance
-
-#### The ssh user has privileges(root/sudo) on RHEL StudioWorkspace Instance / AppDeployment Instance
-
-- No need to do any configurations. Platform will do it automatically.
-
-#### The ssh user don't have privileges(non sudo users) on RHEL StudioWorkspace Instance / AppDeployment Instance
+### Extra configurations on RHEL StudioWorkspace Instance / AppDeployment Instance if ssh user don't have privileges(non sudo users)
 
 - If the user given to the Platform don't have privileged access, then provide below permission for the user given on StudioWorkspace Instance / AppDeployment Instance.
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/on-premise/aws/install-prerequisites.md:201]: not **don't have**

should be

doesn't have

File: learn/on-premise/aws/install-prerequisites.md [hunk 10]
```diff
@@ -101,33 +149,65 @@ Use same version numbers as mentioned.
...
-- Have to execute these commands from privileged user.
-  - Add user to docker group.
-  - Make the user as owner for docker systemd process.
+- Have to execute these commands from a privileged user.
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/on-premise/aws/install-prerequisites.md:202]: Have to execute these command as a privileged user. 

not

**from a**

File: learn/on-premise/aws/install-prerequisites.md [hunk 11]
```diff
@@ -101,33 +149,65 @@ Use same version numbers as mentioned.
...
+  - Add user to the docker group.
+  - Make the user as a owner for the docker systemd process.
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/on-premise/aws/install-prerequisites.md:204]: Replace with:

Make user the owner of the docker `systemd` process

File: learn/on-premise/azure/install-prerequisites.md [hunk 12]
```diff
@@ -9,14 +9,26 @@ sidebar_label: "Install Prerequisites Softwares"
...
+### The ssh user don't have privileges install/upgrade utility software
+
+#### Platform Instance
+
+- No need to do any configurations,WME Installer will automatically install required softwares.
 
-The given ssh user don't have permission to install softwares Then install below as per operating system.
+#### StudioWorkspace Instance / AppDeployment Instance
+
+The given ssh user don't have permission to install software Then install below as per operating system.
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/on-premise/azure/install-prerequisites.md:31]: doesn't 

not 

don't have

File: learn/on-premise/azure/install-prerequisites.md [hunk 13]
```diff
@@ -69,10 +90,37 @@ sudo apt-get install python3 -y
 
 ## RHEL
 
-Install below softwares on Platform Instance and StudioWorkspace Instance / AppDeployment Instance
+### The ssh user has privileges(root/sudo) or user don't have privileges for install/upgrade utility softwares
+
+#### Platform Instance
+
+- If given ssh user has privileges(root/sudo) or user don't have privileges to install/upgrade. WME Installer will automatically install the Docker software. Internet is not required for Installation in this case.
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/on-premise/azure/install-prerequisites.md:97]: Replace this sentence. It does not have any meaning

File: learn/on-premise/azure/install-prerequisites.md [hunk 14]
```diff
@@ -101,33 +149,65 @@ Use same version numbers as mentioned.
...
-- No need to do any configurations. Platform will do it automatically.
-
-#### The ssh user don't have privileges(non sudo users) on RHEL StudioWorkspace Instance / AppDeployment Instance
+### Extra configurations on RHEL StudioWorkspace Instance / AppDeployment Instance if ssh user don't have privileges(non sudo users)
 
 - If the user given to the Platform don't have privileged access, then provide below permission for the user given on StudioWorkspace Instance / AppDeployment Instance.
-- Have to execute these commands from privileged user.
-  - Add user to docker group.
-  - Make the user as owner for docker systemd process.
+- Have to execute these commands from a privileged user.
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/on-premise/azure/install-prerequisites.md:202]: Have to execute these commands as a 
not 
**from** a

File: learn/on-premise/azure/install-prerequisites.md [hunk 15]
```diff
@@ -101,33 +149,65 @@ Use same version numbers as mentioned.
...
+  - Add user to the docker group.
+  - Make the user as a owner for the docker systemd process.
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/on-premise/azure/install-prerequisites.md:204]: Same review comment as above

File: learn/on-premise/gcp/install-prerequisites.md [hunk 16]
```diff
@@ -69,10 +90,37 @@ sudo apt-get install python3 -y
 
 ## RHEL
 
-Install below softwares on Platform Instance and StudioWorkspace Instance / AppDeployment Instance
+### The ssh user has privileges(root/sudo) or user don't have privileges for install/upgrade utility softwares
+
+#### Platform Instance
+
+- If given ssh user has privileges(root/sudo) or user don't have privileges to install/upgrade. WME Installer will automatically install the Docker software. Internet is not required for Installation in this case.
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/on-premise/gcp/install-prerequisites.md:97]: why is there so much duplication of content in these pages?

this sentence occurs in many pages and is wrong in all the places.

File: learn/on-premise/gcp/install-prerequisites.md [hunk 17]
```diff
@@ -101,33 +149,65 @@ Use same version numbers as mentioned.
...
-- No need to do any configurations. Platform will do it automatically.
-
-#### The ssh user don't have privileges(non sudo users) on RHEL StudioWorkspace Instance / AppDeployment Instance
+### Extra configurations on RHEL StudioWorkspace Instance / AppDeployment Instance if ssh user don't have privileges(non sudo users)
 
 - If the user given to the Platform don't have privileged access, then provide below permission for the user given on StudioWorkspace Instance / AppDeployment Instance.
-- Have to execute these commands from privileged user.
-  - Add user to docker group.
-  - Make the user as owner for docker systemd process.
+- Have to execute these commands from a privileged user.
```
Comment (prasanth-reddy-wavemaker-com) [inline learn/on-premise/gcp/install-prerequisites.md:202]: More duplication

**Review by prasanth-reddy-wavemaker-com — CHANGES_REQUESTED** [review]
There are lots of errors. This cannot be merged in the current state. 

Please use grammarly extension in Visual studio code https://marketplace.visualstudio.com/items?itemName=znck.grammarly

There is also duplication of content in lots of pages. Instead of copy pasting common sections of documentation they should be linked.

**Comment by krishnaraoch** [comment]
@prasanth-reddy-wavemaker-com ,  we made the changes.

**Review by krishnaraoch — APPROVED** [review]

**Review by prasanth-reddy-wavemaker-com — APPROVED** [review]

**Pull request merged**